### PR TITLE
chore: k8s - show pending PVCs as stopped icon

### LINKED ISSUE #9322: Applying a PVC that is purposely "pending" does not show
### Bug description

Apply a PVC:

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: task-pv-claim
spec:
  storageClassName: manual
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
```

See that it shows in the CLI:

```sh
▶ k get pvc
NAME                        STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   VOLUMEATTRIBUTESCLASS   AGE
task-pv-claim               Pending                                                                        manual         <unset>                 3m37s
```

It then does not show within the PD k8s UI

### Operating system

mac m1

### Installation Method

None

### Version

1.11.1

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
chore: k8s - show pending PVCs as stopped icon

### What does this PR do?

* Pending PVCs show up as "green" / "active", when instead it should show
as "Stopped" as they are not yet in the Bound phase.
* Based upon the similar interface to Deployments

### Screenshot / video of UI

Screenshot 2024-11-19 at 9 56 40 AM

### What issues does this PR fix or reference?

Closes https://github.com/podman-desktop/podman-desktop/issues/9322

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

Deploy a PVC that is purposely pending:

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: task-pv-claim
spec:
  storageClassName: manual
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
```

See that it's status is "Stopped" in the UI.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: packages/renderer/src/lib/pvc/PVCList.svelte [hunk 1]
```diff
@@ -40,6 +40,7 @@ const pvcUtils = new PVCUtils();
 onMount(() => {
   return kubernetesCurrentContextPersistentVolumeClaimsFiltered.subscribe(value => {
     pvcs = value.map(pvc => pvcUtils.getPVCUI(pvc));
+    console.log('pvcs', pvcs);
```
Comment (axel7083) [inline packages/renderer/src/lib/pvc/PVCList.svelte]:
```suggestion
```

Comment (cdrage) [reply packages/renderer/src/lib/pvc/PVCList.svelte]: Woops!

File: packages/renderer/src/lib/pvc/pvc-utils.ts [hunk 2]
```diff
@@ -29,13 +29,14 @@ export class PVCUtils {
     // PVC's only have "Pending", "Bound" and "Lost" as phases.
     // However, we can also determine if it is in the middle of deleting by
     // checking if the phase is "Terminating" through checking if the object has deletionTimestamp.
+    //
+    // Note: If it is "Pending" it will be considered as "STOPPED" as it is not yet bound, this
+    // is similar to how we handle Deployment statuses.
```
Comment (benoitf) [inline packages/renderer/src/lib/pvc/pvc-utils.ts]: I would prefer I think to have the else if entry, just assigning to status = 'STOPPED' in the else condition.

Comment (benoitf) [reply packages/renderer/src/lib/pvc/pvc-utils.ts]: or is it possible to have PENDING status ?

Comment (cdrage) [reply packages/renderer/src/lib/pvc/pvc-utils.ts]: it is by default set to `let status = 'STOPPED';` (see line 35), so didn't need to add an if statement.

We could add Pending, but  need to make changes to UI component as there is no "PENDING" only starting.

Comment (benoitf) [reply packages/renderer/src/lib/pvc/pvc-utils.ts]: yes it's implicit.

```ts
let status = 'STOPPED'
if (foo = 'bar') { // only if we don't want stopped
 then
}
```

I would prefer explicit way with if / elseif/elseif and else at the end
or using case with every type possible, so if someone introduce a new state, we need to update the function with the mapping being required
 ```ts
let status;
if (foo === '...') {
   status = ...
} else if
} else if

} else {
  // here the default
  status = 'STOPPED'
}
 ```

**Review by benoitf — APPROVED** [review]
suggestion (non blocking)

**Comment by cdrage** [comment]
@benoitf updated with your suggestion

**Review by benoitf — APPROVED** [review]
thanks @cdrage

**Pull request merged**